pkg/cloudencrypt/gcp.go

```diff
@@ -0,0 +1,76 @@
+package cloudencrypt
+
+import (
+	"context"
+
+	kms "cloud.google.com/go/kms/apiv1"
+	"cloud.google.com/go/kms/apiv1/kmspb"
+	"github.com/pkg/errors"
+)
+
+// GCPEncryptor Implements Encryptor using Google Cloud's Key Management Service.
+type GCPEncryptor struct {
+	client *kms.KeyManagementClient
+	keyID  string
+}
+
+func NewGCPEncryptor(ctx context.Context, keyID string) (*GCPEncryptor, error) {
+	client, err := kms.NewKeyManagementClient(ctx)
+	if err != nil {
+		return nil, errors.Wrapf(err, "can't create gpc kms client: %s", err.Error())
+	}
+
+	return &GCPEncryptor{
+		client: client,
+		keyID:  keyID,
+	}, nil
+}
+
+func (encryptor *GCPEncryptor) Encrypt(ctx context.Context, plaintext []byte, metadata ...MetadataKV) ([]byte, error) {
+	additionalData, err := encode(buildMetadataMap(metadata...))
+	if err != nil {
+		return nil, err
+	}
+
+	request := &kmspb.EncryptRequest{
+		Name:                        encryptor.keyID,
+		Plaintext:                   plaintext,
+		AdditionalAuthenticatedData: additionalData,
+	}
+
+	response, err := encryptor.client.Encrypt(ctx, request)
+	if err != nil {
+		return nil, errors.Wrapf(err, "gcp encryption failed: %s", err.Error())
```

Comment:
```suggestion
		return nil, fmt.Sptrinf("gcp encryption failed: %s", err)
```

Reply:
sorry `fmt.Errorf` or `errors.New(fmt.Sprintf())`

Reply:
What's wrong with wrapping though? I'd expect `errors.Wrapf` to be preferred in order to preserve the original error?